### DISCUSSION TITLE
Is there a way to convert TipTap exported HTML to JSON, and vice versa?

### DISCUSSION THREAD

**Discussion by Mansehej** [ueberdosis/tiptap#771] [General Questions] [answered]
I have an application where it is important for me to work with both the HTML as well as JSON versions of any document. Is it possible to convert them on the fly, without setting in the content to a TipTap instance?

Also, is it possible to do this in the backend, with Node.js?

Thanks,
Mansehej.

**Comment by Mansehej** [comment, marked as answer]
Update:
Found another repository which has a parser.
For anyone who stumbles upon this thread, [here's the link to the parser.](https://github.com/ueberdosis/html-to-prosemirror)

**Reply by hanspagel** [reply]
Yes, we built PHP packages to convert ProseMirror JSON to HTML and HTML to ProseMirror JSON. In case someone lands here and wants to do that with Node.js, have a look here: https://discuss.prosemirror.net/t/trying-to-serialize-json-back-to-html-using-node-fromjson-and-domserializer/1512
